### PR TITLE
build: Solve typescript library checks [WEB-894]

### PR DESCRIPTION
**Pull request by ashtonG**
## Description

This makes a few changes to the codebase to ensure that our typescript checks use the library definitions when checking.

## Test Plan

* API behavior in webapp should be unchanged
* `npx tsc` in the `webapp/react` folder should complete without issue

## Commentary (optional)

We update a few DefinitelyTyped types and remove a few types that weren't in use. The InteractiveTable component is modified slightly because the column resize props overlapped with the resize props the antd table type gets from the dom types. We removed the dayjs declaration because it's no longer necessary, and fixed the notebookjs type declaration.

The largest part of this pr is lightly refactoring the typescript code generated from swagger-codegen to not use portable-fetch and also not directly delete the querystring when generating urls in order to fix some oddness stemming from url object behavior. In order to do so, we compile a class that extends the preexisting typescript codegen and ensure it's in the classpath when we generate the types.

## Checklist

- [X] Changes have been manually QA'd
- [X] User-facing API changes need the "User-facing API Change" label.
- [X] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [X] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
[WEB-894]

[WEB-894]: https://determinedai.atlassian.net/browse/WEB-894?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/package.json [hunk 1]
```diff
@@ -96,7 +96,6 @@
     "@types/node": "^18.11.9",
     "@types/react": "^18.0.21",
     "@types/react-dom": "^18.0.8",
-    "@types/react-router-dom": "^5.3.3",
```
Comment (keita-determined) [inline webui/react/package.json:99]: q: is it ok to uninstall this? 
Does `react-router-dom` v6 supports type natively?

File: webui/react/typings/notebook.d.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
 declare module 'notebook' {
-    export default any;
+    const t: any
+    export default t;
```
Comment (mapmeld) [inline webui/react/typings/notebook.d.ts:3]: could you add a comment to explain this notebook module a bit?

File: bindings/templates/src/main/resources/typescript-fetch-globals/modelEnum.mustache [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+ * @export
+ * @enum {string}
+ */
+export enum {{classname}} {
+{{#allowableValues}}
+{{#enumVars}}
+    {{{name}}} = <any> {{{value}}}{{^-last}},{{/-last}}
+{{/enumVars}}
+{{/allowableValues}}
+}
```
Comment (mapmeld) [inline bindings/templates/src/main/resources/typescript-fetch-globals/modelEnum.mustache:12]: FYI: we had a lengthy convo in late 2022 about moving enums to a const -- I would say it's best we get this PR merged and then I'll probably recc. bringing in [this mustache file](https://github.com/determined-ai/determined/pull/5533/files#diff-084ad1859484550b5e7fb2f8ce1e856ba37f627c5618729356155871625d31db)  - (this causes changes across web UI, so better not to roll it in now)

**Review by ClaireNeveu — APPROVED** [review]

File: webui/react/package.json (hunk 1 above)
Comment (ashtonG) [reply webui/react/package.json:99]: Yes -- as of v6, react-router and react-router-dom are written in typescript and distribute their own types. You can check locally in your node_modules.

**Pull request merged**